error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -264,6 +264,32 @@ Flux<T> after(Flux<T> flux, long n) {
     }
   }
 
+  /** Prefer {@link Mono#defaultIfEmpty(Object)} over more contrived alternatives. */
+  static final class MonoSwitchIfEmptyOfMonoJust<T> {
+    @BeforeTemplate
+    Mono<T> before(Mono<T> mono, T object) {
```

Comment:
`object` may be a subtype of `T`, but I see that in either case Refaster matches correctly (didn't commit the test changes, but the point is that this expression is rewritten as well:
```java
Flux.<Object>just("X").switchIfEmpty(Mono.just(1))
```